### PR TITLE
OOP School Library: Set up Association

### PR DESCRIPTION
**Pull request by HammadSiddique**
### In this PR:

- Created a class `Classroom` with the following:
   - `@label` instance variable is initialized in the constructor.
- Created the `has-many/belongs-to` relationship between `Classroom` and `Student`. The following is implemented:
   - Created the `has-many` side (a classroom has many students).
   - Created the `belongs-to` side (a student belongs to a classroom).
- Created a class `Book `with the following:
   - `@title` and `@author` instance variables, are initialized in the constructor.
- Created a class `Rental` with the following:
   - `@date` instance variable, should be initialized in the constructor.
- Created the `many-to-many` (also has-many-through) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
   - Created the `has-many` side of `Book` and `Rental` (a book has many rentals).
   - Created the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
   - Created the `has-many` side of `Person` and `Rental` (a person has many rentals).
   - Created the belongs-to side of `Rental` and `Person` (a rental belongs to a person).
   - Modified the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author
+
```
Comment (uwadonat) [inline book.rb]: Let's make `rentals` also available in `attr_accessor`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (uwadonat) [inline book.rb]: - [x] `add_rental method should receive `person` and `date` as parameter. Also the book itself should be added as parameter.
`
def add_rental(person, date)
...
end
`

File: person.rb [hunk 3]
```diff
@@ -27,4 +28,8 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (uwadonat) [inline person.rb]: - [x] `add_rental method should receive `book` and `date` as parameter. Also the person itself should be added as parameter.
`
def add_rental(book, date)
...
end
`

**Review by uwadonat — CHANGES_REQUESTED** [review]
## Hi @HammadSiddique,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Well organized and clean project ✔️
- Professional readme file ✔️
- Descriptive pull request message ✔
almoo
️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (HammadSiddique) [reply book.rb]: Done

File: book.rb (hunk 2 above)
Comment (HammadSiddique) [reply book.rb]: Done

File: person.rb (hunk 3 above)
Comment (HammadSiddique) [reply person.rb]: Done

**Review by roshan-bajgain — APPROVED** [review]
Hi @HammadSiddique, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- PR has a good title and summary. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**